src/interfaces/blobs.js

```diff
@@ -364,25 +371,33 @@ export default function ({url}) {
 						state: BLOB_STATE.ABORTED
 					};
 				case transformationFailed:
+					logger.log('debug', `case: ${op}, Error: ${payload.error}`);
 					return {
 						state: BLOB_STATE.TRANSFORMATION_FAILED,
 						modificationTime: moment(),
 						$set: {
 							'processingInfo.transformationError': payload.error
 						}
 					};
-				case transformationDone:
-					if ('numberOfRecords' in payload) {
+				case transformedRecord:
+					if (payload.transformedRecord !== null) {
```

Comment:
There is content only when the transformation failed for a single record. Change the property name in the body to be more appropriate (e.g. `payload.transformedRecord` -> `payload.error`).

Also, do not check for null, but for `undefined`. But this can be written as follows `if (payload.transformedRecord) {`